### PR TITLE
Add get_latest_ledger_infos functionality to storage service.

### PR DESCRIPTION
**Pull request by dmitri-perelman**
Summary:
Ledger store has been already keeping ledger infos keyed by their epoch number, hence it could already return all the latest ledger infos per epoch.
In this diff we're merely exposing this functionality all the way up to the storage client (which requires adding the protobuf function for the storage service).
This is a purely technical change that doesn't involve any change in the business logic: no one is calling the `get_latest_ledger_infos()` function yet.

Testing: existing unit test coverage

Issue: ref #849

### CODE REVIEW

File: storage/libradb/src/lib.rs [hunk 1]
```diff
@@ -300,6 +300,20 @@ impl LibraDB {
             .version())
     }
 
+    /// Return the latest ledger infos per epoch starting with the given epoch num:
```
Comment (msmouse) [inline storage/libradb/src/lib.rs]:
```suggestion
    /// Returns the latest ledger infos per epoch starting with the given epoch num:
```

File: storage/libradb/src/ledger_store/mod.rs [hunk 2]
```diff
@@ -61,12 +61,11 @@ impl LedgerStore {
...
     }
 
-    /// Return the ledger infos with their least 2f+1 signatures starting from `start_version` to
+    /// Return the ledger infos with their least 2f+1 signatures starting from `start_epoch` to
     /// the most recent one.
     /// Note: ledger infos and signatures are only available at the last version of each earlier
     /// epoch and at the latest version of current epoch.
-    #[cfg(test)]
-    fn get_ledger_infos(
+    pub fn get_latest_ledger_infos_per_epoch(
```
Comment (msmouse) [inline storage/libradb/src/ledger_store/mod.rs]: the rename seems unnecessary? (especially when you still call it the old way in `storage_client` etc.

File: storage/storage_proto/src/lib.rs [hunk 3]
```diff
@@ -371,6 +371,99 @@ impl IntoProto for GetStartupInfoResponse {
...
+}
+
+impl GetLatestLedgerInfosPerEpochRequest {
+    /// Constructor.
+    pub fn new(start_epoch: u64) -> Self {
+        Self { start_epoch }
+    }
+}
+
+impl FromProto for GetLatestLedgerInfosPerEpochRequest {
```
Comment (msmouse) [inline storage/storage_proto/src/lib.rs]: We should be able to derive `FromProto`/`IntoProto`?

File: storage/storage_proto/src/proto/storage.proto [hunk 4]
```diff
@@ -113,3 +117,13 @@ message StartupInfo {
...
+
+message GetLatestLedgerInfosPerEpochRequest {
+    /// The last epoch number with available information to the client.
+    uint64 start_epoch = 1;
+}
+
+message GetLatestLedgerInfosPerEpochResponse {
+    /// Vector of latest ledger infos per epoch (not sorted)
+    repeated types.LedgerInfoWithSignatures latest_ledger_infos = 1;
+}
```
Comment (msmouse) [inline storage/storage_proto/src/proto/storage.proto]: probably add the end of line?

File: storage/libradb/src/ledger_store/mod.rs (hunk 2 above)
Comment (dmitri-perelman) [reply storage/libradb/src/ledger_store/mod.rs]: The name get_ledger_infos() wasn't very intuitive to me. What about get_latest_ledger_infos()? We could go to get_latest_ledger_infos_per_epoch(), that would make it even more intuitive, but I don't have strong opinions there...

File: storage/storage_proto/src/lib.rs (hunk 3 above)
Comment (dmitri-perelman) [reply storage/storage_proto/src/lib.rs]: I initially tried automated derivation but the compiler didn't like it, let me try again :)

**Comment by dmitri-perelman** [comment]
Updates following @msmouse comments

File: storage/libradb/src/ledger_store/mod.rs (hunk 2 above)
Comment (lightmark) [reply storage/libradb/src/ledger_store/mod.rs]: `get_last_ledger_infos()`?

**Comment by msmouse** [comment]
I'm fine with either name as long as it's consistent.

@bors-libra r+
